packages/react-lowcode/src/codegen/generation/generators/index/table-generator-factory.ts

```diff
@@ -4,28 +4,32 @@ import GenerationContext from '../../context'
 import {BasicTableGenerator} from './basic-table-generator'
 import MuiDataTableGenerator from './mui-dt-generator'
 import GrommetDataTableGenerator from './grommet-dt-generator'
+import { Entity } from '../../entity';
 
 export interface TableGenerator{
     generateTableComponent(): PageComponent;
 }
 
 export class TableGeneratorFactory{
     private _context: GenerationContext;
+    private _entity: Entity;
 
-    constructor(context: GenerationContext){
+
+    constructor(context: GenerationContext, entity: Entity){
         this._context = context;
+        this._entity = entity;
     }
 
     getTableGenerator(): TableGenerator{
-        let generator: TableGenerator = new BasicTableGenerator(this._context);
+        let generator: TableGenerator = new BasicTableGenerator(this._context, this._entity);
 
-        if(this._context.tableType === TableType.DataTable) {
+        if(this._context.index?.tableType === TableType.DataTable) {
             switch(this._context.uiFramework){
```

Comment:
instead of having map we could use map with default instances representingu mui and grommet